text/0000-cfg-path-version.md

```diff
@@ -0,0 +1,484 @@
+For example:
+
+```rust
+#[cfg_attr(version("1.27"), must_use)]
+fn double(x: i32) -> i32 {
+    2 * x
+}
+
+fn main() {
+    double(4);
+    // warning: unused return value of `double` which must be used
+    // ^--- This warning only happens if we are on Rust >= 1.27.
+}
+```
+
+Another example is opting into the system allocator on Rust 1.28 and beyond:
+
+```rust
+#[cfg(version("1.28"))
+// or: #[cfg(path_exists(::std::alloc::System))]
+use std::alloc::System;
+
+#[cfg_attr(version("1.28"), global_allocator)]
+static GLOBAL: System = System;
+
+fn main() {
+    let mut v = Vec::new();
+    // This will allocate memory using the system allocator.
+    // ^--- But only on Rust 1.28 and beyond!
+    v.push(1);
+}
+```
+
+Note that you won't be able to make use of `#[cfg(version(..))` for these 
+particular features since they were introduced before this RFC's featuers
+get stabilized. However, there will be features in the future to use this
+mechanism on.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+## `#[cfg(nightly)]`
+
+To the `cfg` attribute , a `nightly` flag is added.
+
+If and only if a Rust compiler considers itself to be on a nightly channel
+it the `nightly` flag be considered active.
+
+## `#[cfg(version("<version_string>"))]`
+
+To the `cfg` attribute, a `version` flag is added.
+This flag requires that a string literal be specified in it inside parenthesis.
+The string literal must have the format:
+
+```
+version_string ::= \d(.\d)?
+```
+
+If and only if a Rust compiler considers itself to have a version which is
+greater or equal to the version in the `version_string` will the
+`#[cfg(version("<version_string>")]` flag be considered active.
+
+## `#[cfg(path_exists($path))]`
+
+To the `cfg` attribute, a `path_exists` flag is added.
+This flag requires that a `path` fragment be specified in it inside parenthesis
+but not inside a string literal. The `$path` must start with leading `::`
+and may not refer to any parts of the own crate (e.g. with `::crate::foo`,
+`::self::foo`, or `::super::foo` if such paths are legal).
+This restriction exists to ensure that the user does not try to
+conditionally compile against parts of their own crate because that crate
+has not been compiled when the `path_exists` flag is checked on an item.
+
+If and only if the path referred to by `$path` does exist will the
+`#[cfg(path_exists($path)]` flag be considered active.
+In checking whether the path exists or not, the compiler will consider
+feature gated items to exist if the gate has been enabled.
+If a path refers to an item inside an inherent implementation,
+the path will be considered to exist if any configuration of generic
+parameters can lead to the item. To check whether an item exists for
+an implementation with a specific sequence of concrete types applied to
+a type constructor, it is possible to use the `::foo::bar::<T>::item` syntax.
```

Comment:
Re. `where` clauses I'd say no... the path grammar doesn't accept it, but I think it would also be too much for an initial version of this.

Re. implementation I'll note it as an unresolved question.